homework/list.js

```diff
@@ -50,30 +57,35 @@
       // Note: buff is captured, so we now have a
       // little closure naturally.
       buff.appendChild(li);
-    });
+    }).bind(this));
     ul.appendChild(buff);
     _wrapper.appendChild(ul);
   }
 
-  function fetchList(afterFetch) {
-    var xhr = new XMLHttpRequest();
-    xhr.open('GET', 'http://127.0.0.1:8000/demo-list-notes.json', true);
-    xhr.responseType = 'json';
-    xhr.onreadystatechange = function(e) {
-      // Watch out: we have a mysterious unknown 'this'.
-      if (this.readyState === 4 && this.status === 200) {
-        var listData = this.response;
-        // The flow ends here.
-        afterFetch(listData);
-      } else if (this.status !== 200 ){
-        // Ignore error in this case.
-      }
-    };
-    xhr.send();
+  list.prototype.fetchList = function(afterfetch) {
+    return new Promise((function(resolve, reject) {//use Promise
+		var xhr = new XMLHttpRequest();
+		xhr.open('GET', 'http://127.0.0.1:8000/demo-list-notes.json', true);
+		xhr.responseType = 'json';
+		xhr.onreadystatechange = function(e) {
+		// Watch out: we have a mysterious unknown 'this'.
+			if (this.readyState === 4 && this.status === 200) {
+				var listData = this.response;
+				// The flow ends here.
+				resolve(listData);
+			} 
+			else if (this.status !== 200 ){
+				reject('FETCHING FAILED: ' + this.status + ' ' + this.readyState);
+				// Ignore error in this case.
+			}
+		};
+		xhr.send();
+	}).bind(this));
   }
 
   document.addEventListener('DOMContentLoaded', function(event) {
     start();
   });
+  module.exports.add = add;
```

Comment:
Did you really test your code? And what made you feel your code is with **good expressiveness and consistent convention** to show your skills? Please, give your work more thoughts. If you have any issue you can't deal with, **ask me**. The only thing worse than making a mistake is that you keep in silent and do not ask for help.